glean-core/android/src/test/java/mozilla/telemetry/glean/private/EventMetricTypeTest.kt

```diff
@@ -524,4 +525,63 @@ class EventMetricTypeTest {
         ping.submit()
         assertNull(waitForPingContent(pingName, null, server))
     }
+
+    @Test
+    fun `Event listener correct reports event recordings`() {
+        // Define some click events to listen for in the listener
+        val click1 = EventMetricType<ClickExtras>(
+            CommonMetricData(
+                disabled = false,
+                category = "ui",
+                lifetime = Lifetime.PING,
+                name = "click1",
+                sendInPings = listOf("store1"),
+            ),
+            allowedExtraKeys = listOf(),
+        )
+        val click2 = EventMetricType<ClickExtras>(
+            CommonMetricData(
+                disabled = false,
+                category = "ui",
+                lifetime = Lifetime.PING,
+                name = "click2",
+                sendInPings = listOf("store1"),
+            ),
+            allowedExtraKeys = listOf(),
+        )
+        val click3 = EventMetricType<ClickExtras>(
+            CommonMetricData(
+                disabled = false,
+                category = "ui",
+                lifetime = Lifetime.PING,
+                name = "click3",
+                sendInPings = listOf("store1"),
+            ),
+            allowedExtraKeys = listOf(),
+        )
+
+        class TestEventListener : GleanEventListener {
+            var id: String = ""
```

Comment:
would probably name this `lastSeenId` or `lastId` to make it obvious that we overwrite this with every call.